osu.Game.Tournament/IPC/FileBasedIPC.cs

```diff
@@ -89,9 +85,9 @@ private Storage initialiseIPCStorage(string path)
 
                                     lastBeatmapId = beatmapId;
 
-                                    var existing = ladder.CurrentMatch.Value?.Round.Value?.Beatmaps.FirstOrDefault(b => b.ID == beatmapId && b.Beatmap != null);
+                                    var existing = ladder.CurrentMatch.Value?.Round.Value?.Beatmaps.FirstOrDefault(b => b.ID == beatmapId);
 
-                                    if (existing != null)
+                                    if (existing?.Beatmap != null)
```

Comment:
This changes population logic. A beatmap with an invalid ID that cannot be fetched will be fetched over and over again if it pops up in several rounds rather than once.

Also this possibly breaks flows where the stable ipc changes to no beatmap or something. `MatchIPCInfo.Beatmap` likely needs to be nullable too.

Reply:
but it need to add null check in `beatmapChanged` for other component, it that ok?

Reply:
Which "other component"? I don't understand.

Reply:
I mean https://github.com/ppy/osu/blob/master/osu.Game.Tournament/Screens/MapPool/MapPoolScreen.cs#L126-L134 and
https://github.com/ppy/osu/blob/master/osu.Game.Tournament/Screens/BeatmapInfoScreen.cs#L40-L44

Reply:
Well if you look one step further, `SongBar` is already checking for null beatmap? Also `#nullable disable` is still in that file. Was this PR intending to remove it everywhere in the tournament project?

Reply:
songbar is still `nullable disable`. please provide reasoning.

if in the next review pass i find more issues i'm going to either stop reviewing and close, or reapply myself. the overhead of reviewing stupid stuff like this is too high if the contributor isn't doing a good job.